app/routes/languages.rb

```diff
@@ -41,6 +44,23 @@ class Languages < Core
         end
       end
 
+      get '/languages/:track_id/:topic' do |track_id, topic|
+        return language_not_found(topic) unless TOPICS.include?(topic.to_sym)
```

Comment:
I looked at the code again and I think that other than changing the method name we should be good on all cases. 

```ruby
return topic_not_found(topic) unless TOPICS.include?(topic.to_sym)

_, body = X::Xapi.get('tracks', track_id)
parsed_body = JSON.parse(body)
if parsed_body['error'] == "No track '#{track_id}'"
  language_not_found(track_id)
else
  # code that loads the page
```

- return early if topic does not exist
  - continue if it does
    - check track
      - return 404 if it does not
      - if track is valid render page. (at this point we have already validated both the track and topic)

From what I understand we will not have a situation where there will be a valid topic with an invalid track rendering track page.